velox/docs/functions/spark/datetime.rst

```diff
@@ -70,6 +70,12 @@ These functions support TIMESTAMP and DATE input types.
 
     This is an alias for :func:`day_of_week`.
 
+.. spark:function:: hour(timestamp) -> integer
+
+    Returns the hour of ``timestamp``. ::
```

Comment:
Would you clarify the type of 'timestamp'? Is it TIMESTAMP or VARCHAR? If the latter, what are supported formats?

Reply:
Would you also clarify how the results are affected by the session timezone? Please, include an example.

Reply:
Sorry for late reply, after second thought and check spark code again, spark hour function takes both timestamp and string type as args, so here we need accept string as argType.

For now, Velox `Timestamp` doesn't impl `parseTo` function, i'd like to keep the PR and do the conversion from string to timestamp first.

https://github.com/facebookincubator/velox/blob/8f0dd8b72240b48f77bfa98a2ad326e845d6879c/velox/type/Timestamp.cpp#L321-L323

Thanks for your review! And if you have any suggestions in this regard, I would be happy to hear.

Reply:
Hi @zwangsheng, it seems hour function only accepts Timestamp type (TimestampNTZ is also allowed, but we can ignore it for now). See [spark source code](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L376).
For String input, spark will implicitly plan a cast function to get Timestamp intermediate result. And I think velox already supports casting String to Timestamp cc @rui-mo. Anyway, for this PR, we can just consider Timestamp type since String type cannot be passed to it.

Reply:
Thanks for the inform! 😃 I will look thought and finish this.